controllers/workspace/eventhandlers.go

```diff
@@ -47,7 +47,13 @@ func dwRelatedPodsHandler(obj client.Object) []reconcile.Request {
 }
 
 func (r *DevWorkspaceReconciler) dwPVCHandler(obj client.Object) []reconcile.Request {
-	// Check if PVC is owned by a DevWorkspace (per-workspace storage case)
+	if obj.GetDeletionTimestamp() == nil {
+		return []reconcile.Request{}
```

Comment:
I'm slightly hesitant to prevent reconciles unless the PVC was deleted (i.e. ignoring modifications to a per-workspace PVC), just because @amisevsk might have had some edge case in mind when he made https://github.com/devfile/devworkspace-operator/commit/2a44ff6da5aa079edd1f21b4032bb9aba10600c2:

> Update the watch for PVCs to also enqueue reconciles for per-workspace
> PVC events. Previously, the per-workspace PVC could be **modified**/deleted
> without the DevWorkspace that owns that PVC noticing.

However:
- We weren't queuing reconciles for common/per-user PVCs unless they were being deleted, so there's an inconsistency between per-workspace PVCs & per-user PVCs causing reconciles upon modification
- Users are **not** supposed to **modify** PVCs created by DWO. Attempting to do so in the OpenShift UI will present the user with a warning: `This resource is managed by DevWorkspaceDW
<DevWorkspace Name> and any modifications may be overwritten. Edit the managing resource to preserve changes.`. **However:** editing the PVC with kubectl/oc will **not give any warning.**
- As discussed in point 1. of my earlier comment https://github.com/devfile/devworkspace-operator/pull/1233#discussion_r1508110341: PVCs are currently considered immutable from DWO's perspective. Any modification made to a DWO-owned PVC cannot be automatically amended by DWO, so reconciling on modification doesn't currently yield any advantages. **However:** in the future, if we resolve https://github.com/devfile/devworkspace-operator/issues/1250 and allow for DWO to modify PVCs to add the `controller.devfile.io/devworkspace_pvc_type` label, then a user removing this label _should_ trigger a reconcile to allow DWO to re-apply the label. This is a problem for the future however...


In summary, I think this optimization is okay :man_shrugging: I just wanted to write out these thoughts beforehand.